src/ws/test-server.c

```diff
@@ -504,11 +510,34 @@ on_handle_resource (CockpitWebServer *server,
   inject_address (response, "bus_address", bus_address);
   inject_address (response, "direct_address", direct_address);
 
-  cockpit_web_response_set_cache_type (response, COCKPIT_WEB_RESPONSE_NO_CACHE);
   cockpit_web_response_file (response, rebuilt,  cockpit_web_server_get_document_roots (server));
+  g_free (rebuilt);
+}
+
+static gboolean
+on_handle_resource (CockpitWebServer *server,
+                    const gchar *path,
+                    GHashTable *headers,
+                    CockpitWebResponse *response,
+                    gpointer user_data)
+{
+  gchar **parts;
+
+  g_assert (g_str_has_prefix (path, "/pkg"));
+
+  cockpit_web_response_set_cache_type (response, COCKPIT_WEB_RESPONSE_NO_CACHE);
+
+  parts = g_strsplit (path, "/", -1);
+  if (g_strcmp0 (parts[2], "manifests.js") == 0 && !parts[3])
```

Comment:
I think that should be `manifest.json`.

Is path just `pkg/manifest.json`? If it is something like `/pkg/base1/manifest.json`, it would need to be `parts[3]`.